### PR TITLE
PR to merge Rspec to master

### PR DESCRIPTION
**Pull request by Haroonabdulrazaq**
- Add Test to my_each
- Add Test to my_each_with_index
- Add Test to my_count
- Add Test to my_inject
- Add Test to my_select
- Add Test to my_any?
- Add Test to my_all?
- Add Test to my_none?
- Add Test to my_map

### CODE REVIEW

File: spec/enumerables_spec.rb [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+        expect(my_result).to eql(false)
+      end
+      it 'returns false if any element matches the class' do
+        my_result = bool_arr.my_none?(TrueClass)
+        expect(my_result).to eql(false)
+      end
+      it 'returns false if any element is true and argument is nil' do
+        my_result = arr.my_none?
+        expect(my_result).to eql(false)
+      end
```
Comment (maelfosso) [inline spec/enumerables_spec.rb]: I tweak this test a bit and I got an error
```rb
my_result = arr.my_none?(String)
expect(my_result).to eql(true)
```

Comment (maelfosso) [reply spec/enumerables_spec.rb]: And I also did that changes and I got an error
```rb
my_result = word_arr.my_none?(/z/)
expect(my_result).to eql(word_arr.none?(/z/))
```

File: spec/enumerables_spec.rb [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+        expect(my_result).to eql(true)
+      end
+      it 'returns true if any element matches the class' do
+        my_result = bool_arr.my_any?(TrueClass)
+        expect(my_result).to eql(true)
+      end
+      it 'returns true if any element is true and argument is nil' do
+        my_result = arr.my_any?
+        expect(my_result).to eql(true)
+      end
```
Comment (maelfosso) [inline spec/enumerables_spec.rb]: Only **true**? Add more tests for **falsy** situation

File: spec/enumerables_spec.rb [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+        hash.each { |key| original_hash << key }
+        expect(my_hash).to eql(original_hash)
+      end
+    end
+
+    context 'If block not given' do
+      it 'returns enum if block is not given' do
+        expect(arr.my_each.is_a?(Enumerable)).not_to be false
+      end
+    end
```
Comment (maelfosso) [inline spec/enumerables_spec.rb:36]: What about **truthy** situation?

**Review by maelfosso — CHANGES_REQUESTED** [review]
# Changes required :red_circle: 

Hello,
Thank you for the effort done is this project.
To move forward, there are some changes to make

1. Kindly, read the comments left below

2. In the tests, add more tests for **String**, **patterns**, **Integer**

3. Also, add tests where you expect **true** and **false**. 

Kindly make the changes and submit a new code review requests
Thanks

Mael FOSSO

File: spec/enumerables_spec.rb [hunk 4]
```diff
@@ -0,0 +1,234 @@
+require './enumerables.rb'
```
Comment (christianotieno) [inline spec/enumerables_spec.rb:1]:
```suggestion
require '../enumerables.rb'
```
- A period was missing in this file reference. kindly make sure you add that. Thank you

**Review by christianotieno — APPROVED** [review]
# PROJECT APPROVED :champagne:
Excellent job on this project guys. :fireworks: Keep it up.
##
***All the required tasks have been fulfilled in this project. I will go ahead and mark this as complete.***
- Congratulations!
- Before you merge. Kindly take note on the minor suggested change I highlighted below 👇 
- You don't have to submit another CRR after this. You can then merge to master.
- I wish you all the very best in your next tasks 🥇 
##
Happy coding :smile:
**[Christian Otieno](https://github.com/ChristianOtieno)**
[Twitter](https://twitter.com/iamchrisotieno) [linkedin](https://www.linkedin.com/in/christianotieno/) [GitLab](https://gitlab.com/christianotieno)

**Pull request merged**